File: cctbx/cctbx/sgtbx/lattice_symmetry.py
from cctbx import sgtbx

class group_search(object):

  def __init__(self, modulus=2):
    self.ext = sgtbx.lattice_symmetry_group_search(modulus)

  def n_potential_axes(self):
    return self.ext.n_potential_axes()

  def __call__(self,
        minimum_cell,
        max_delta=3.,
        enforce_max_delta_for_generated_two_folds=True):
    return self.ext(
      minimum_cell, max_delta, enforce_max_delta_for_generated_two_folds)

group = group_search()

def find_max_delta(minimum_cell, group):
  return sgtbx.lattice_symmetry_find_max_delta(minimum_cell, group)

File: cctbx/sgtbx/boost_python/lattice_symmetry.cpp
#include <boost/python/def.hpp>
#include <boost/python/args.hpp>
#include <cctbx/sgtbx/lattice_symmetry.h>

namespace cctbx { namespace sgtbx { namespace boost_python {

  void wrap_lattice_symmetry()
  {
    using namespace boost::python;

    def("lattice_symmetry_find_max_delta", lattice_symmetry::find_max_delta, (
      arg("reduced_cell"), arg("space_group")));

    def("lattice_symmetry_group", lattice_symmetry::group, (
      arg("reduced_cell"),
      arg("max_delta")=3.,
      arg("enforce_max_delta_for_generated_two_folds")=true));
  }

}}} // namespace cctbx::sgtbx::boost_python
